src/monitor/src/main/java/ee/ria/xroad/monitor/MetricsProviderActor.java

```diff
@@ -50,51 +48,95 @@
 
     private LoggingAdapter log = Logging.getLogger(getContext().system(), this);
 
+    /**
+     * Two phase filter for checking user requested metric names and additional chained filter for
+     * application defined metric names (histogram/process/certificate/package/etc).
+     */
+    public class SystemMetricsFilter implements MetricFilter {
+
+        private List<String> metricNames;
+        private MetricFilter chainedFilter;
```

Comment:
The fields should be final, if only set in object construction

Reply:
Corrected.